lib/variable/multiply.cpp

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+#include "scipp/variable/multiply.h"
+#include "scipp/core/dtype.h"
+#include "scipp/core/element/arithmetic.h"
+#include "scipp/variable/transform.h"
+
+namespace scipp::variable {
+
+Variable operator*(const Variable &a, const Variable &b) {
+  if (a.dtype() == dtype<Eigen::Affine3d>) {
```

Comment:
Should this check `a` as well as `b`?

Reply:
I think at the moment we are only ever planning to support `transform * vector` and `transform * transform`, not `vector * transform`, so this seems sufficient?

Reply:
There will be `translation * transform`, `transform * translation`, ...

Reply:
Those other types haven't been added yet but I assumed that this would just be `a.dtype = transform || a.dtype = translation || a.dtype = rotation || ...`, i.e. we never care what dtype `b` is, only that `a` is one of the transform types?

Reply:
Well, e.g., linear transformations should not have a unit, so they should not be handled by this branch. I think this is the case in general: Unless *both* `a` and `b` contain a translation (or are a translation) we do not want to enter this branch?